### PR TITLE
Restart proxies

### PR DESCRIPTION
**Pull request by triffer**
**Description**

Changes proposed in this pull request:

- Restart proxies if proxy configuration changed in Istio CR or in bundled default Istio Operator
- Decrease error reconciliation time
- Restructure integration tests and make them independent
- Remove CNI proxy restart

**Related issue(s)**

- https://github.com/kyma-project/istio/issues/152

### CODE REVIEW

File: pkg/lib/sidecars/test/features/update.feature [hunk 1]
```diff
@@ -20,10 +20,46 @@ Feature: Istio upgrade
...
+
+  Scenario: Updating CPU requests Proxy configuration
+    Given there is a cluster with Istio "1.14.4", default injection == "true"
+    And there are Pods with Istio "1.14.4" sidecar and resource "requests" cpu "10m" and memory "100Mi"
+    When a restart happens with target Istio "1.14.4" and sidecar resource "requests" set to cpu "20m" and memory "100Mi"
+    Then no resource that is not supposed to be restarted is restarted
+    And no resource that is not supposed to be deleted is deleted
+    And all required resources are restarted
+    And all required resources are deleted
+
```
Comment (barchw) [inline pkg/lib/sidecars/test/features/update.feature:39]: Should we check the resources afterwards?
```suggestion
And there are Pods with Istio "1.14.4" sidecar and resource "requests" cpu "20m" and memory "100Mi"

```

File: pkg/lib/sidecars/test/features/update.feature [hunk 2]
```diff
@@ -20,10 +20,46 @@ Feature: Istio upgrade
...
+  Scenario: Updating memory requests Proxy configuration
+    Given there is a cluster with Istio "1.14.4", default injection == "true"
+    And there are Pods with Istio "1.14.4" sidecar and resource "requests" cpu "10m" and memory "100Mi"
+    When a restart happens with with target Istio "1.14.4" and sidecar resource "requests" set to cpu "10m" and memory "200Mi"
+    Then no resource that is not supposed to be restarted is restarted
+    And no resource that is not supposed to be deleted is deleted
+    And all required resources are restarted
+    And all required resources are deleted
+
```
Comment (barchw) [inline pkg/lib/sidecars/test/features/update.feature:48]:
```suggestion
And there are Pods with Istio "1.14.4" sidecar and resource "requests" cpu "20m" and memory "100Mi"

```

File: pkg/lib/sidecars/test/features/update.feature [hunk 3]
```diff
@@ -20,10 +20,46 @@ Feature: Istio upgrade
...
+  Scenario: Updating cpu limits Proxy configuration
+    Given there is a cluster with Istio "1.14.4", default injection == "true"
+    And there are Pods with Istio "1.14.4" sidecar and resource "limits" cpu "100m" and memory "200Mi"
+    When a restart happens with target Istio "1.14.4" and sidecar resource "limits" set to cpu "200m" and memory "200Mi"
+    Then no resource that is not supposed to be restarted is restarted
+    And no resource that is not supposed to be deleted is deleted
+    And all required resources are restarted
+    And all required resources are deleted
+
```
Comment (barchw) [inline pkg/lib/sidecars/test/features/update.feature:57]:
```suggestion
And there are Pods with Istio "1.14.4" sidecar and resource "limits" cpu "200m" and memory "200Mi"

```

File: pkg/lib/sidecars/test/features/update.feature [hunk 4]
```diff
@@ -20,10 +20,46 @@ Feature: Istio upgrade
...
+  Scenario: Updating cpu limits Proxy configuration
+    Given there is a cluster with Istio "1.14.4", default injection == "true"
+    And there are Pods with Istio "1.14.4" sidecar and resource "limits" cpu "100m" and memory "200Mi"
+    When a restart happens with target Istio "1.14.4" and sidecar resource "limits" set to cpu "100m" and memory "300Mi"
+    Then no resource that is not supposed to be restarted is restarted
+    And no resource that is not supposed to be deleted is deleted
+    And all required resources are restarted
     And all required resources are deleted
```
Comment (barchw) [inline pkg/lib/sidecars/test/features/update.feature:65]:
```suggestion
    And all required resources are deleted
    And there are Pods with Istio "1.14.4" sidecar and resource "limits" cpu "100m" and memory "300Mi"

```

**Review by barchw — CHANGES_REQUESTED** [review]

**Review by barchw — CHANGES_REQUESTED** [review]

File: pkg/lib/sidecars/test/features/update.feature (hunk 1 above)
Comment (triffer) [reply pkg/lib/sidecars/test/features/update.feature:39]: As discussed, those are the mocked sidecar tests that don't updated the resources, so we don't want to verify it here and it's already verified in the integration tests.

**Review by barchw — APPROVED** [review]

**Pull request merged**